libnetwork/driver_plugin.py

```diff
@@ -136,24 +136,34 @@ def request_pool():
     sub_pool = json_data.get("SubPool")
     v6 = json_data["V6"]
 
-    # Calico IPAM does not allow you to request Pool or SubPool.
-    if pool or sub_pool:
-        error_message = "Calico IPAM does not support pool configuration on " \
-                        "'docker create network'.  Calico IP Pools should " \
-                        "be configured first and IP assignment is from " \
-                        "those pre-configured pools."
+    # Calico IPAM does not allow you to request SubPool.
+    if sub_pool:
+        error_message = "Calico IPAM does not support sub pool configuration" \
+                        "on 'docker create network'.  Calico IP Pools " \
+                        "should be configured first and IP assignment is " \
+                        "from those pre-configured pools."
         app.logger.error(error_message)
         raise Exception(error_message)
 
-    # We use static pool ID and CIDR - these are arbitrary since we do not
-    # actually create a pool here.  The user should have configured a Calico
-    # IP pool which will be used for address assignment.
+    # If a pool (subnet on the CLI) is specified, it must match one of the
+    # preconfigured Calico pools.
+    if pool:
+        if not get_pool(IPNetwork(pool)):
+            error_message = "The subnet on the 'docker create network' " \
```

Comment:
What if they're using the docker rest API. This error should be a little more generic